tests/function-controller/testsuite/testsuite.go

```diff
@@ -52,9 +52,11 @@ func New(restConfig *rest.Config, cfg Config, t *testing.T, g *gomega.GomegaWith
 		return nil, errors.Wrap(err, "while creating K8s Dynamic client")
 	}
 
-	ns := namespace.New(coreCli, cfg.Namespace)
-	f := newFunction(dynamicCli, cfg.FunctionName, cfg.Namespace, cfg.WaitTimeout, t.Logf)
-	ar := apirule.New(dynamicCli, cfg.APIRuleName, cfg.Namespace, cfg.WaitTimeout, t.Logf)
+	namespaceName := fmt.Sprintf("%s-%d", cfg.Namespace, rand.Uint32())
```

Comment:
oh, is that a reason you've deleted first cleanup? new namespace + hostname every time?